### PR TITLE
Magicians Bookstore Project: Fourth Milestone

### PR DESCRIPTION
**Pull request by SmithVyne**
I implemented all the features of this project according to the [instructions here](https://microverse.pathwright.com/library/react-redux-v11/184438/path/step/106376030/)

### CODE REVIEW

File: src/reducers/filter.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const filterReducer = (state = '', action) => {
```
Comment (whiz25) [inline src/reducers/filter.js:1]: - assign the category `All` as default. `const filterReducer = (state = 'All', action)`

File: src/components/CategoryFilter.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+const CategoryFilter = ({ filterChange }) => (
+  <select onChange={e => filterChange(e.target.value)}>
+    <option disabled selected>Filter Categories</option>
+    <option value="">All</option>
+    {
+      Categories.map(category => (<option key={category}>{category}</option>))
+    }
+  </select>
+);
```
Comment (whiz25) [inline src/components/CategoryFilter.js:12]: - merge the category `All` with available categories.
```javascript
const filterCategories = [...['All']...Categories]
filterCategories.map((category){})
```

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Nice work on this PR. Let's fix some things before merging.
Check my comments for suggestions.
- [x] add a title to this PR.


### Martin Afani

**Review by Torres-ssf — APPROVED** [review]
### PR Approved
Hi @SmithVyne :hand: 
Awesome job implementing the requested changes.
You may merge the PR
Good luck with the next milestone

Happy coding :computer: 
**Sergio Torres**

**Pull request merged**